File: internal/model/battle.go
package model

import (
	"time"

	"github.com/google/uuid"
)

type BattleModel struct {
	ID              uuid.UUID
	AttackerID      uuid.UUID
	AttackerName    string
	DefenderID      uuid.UUID
	DefenderName    string
	DamageDealt     int
	WinnerID        *uuid.UUID
	WinnerName      string
	AttackerHPFinal int
	DefenderHPFinal int
	RoundsCount     int
	RoundsData      string // JSON armazenado no banco
	CreatedAt       time.Time
}

File: frontend/src/types/battle.ts
export interface BattleCreateRequest {
    attacker_character_id: string;
    defender_character_id: string;
}

export interface BattleRound {
    round: number;
    attacker_name: string;
    defender_name: string;
    damage: number;
    remaining_hp: number;
    is_critical: boolean;
    message: string;
}

export interface Battle {
    id: string;

    attacker_id: string;
    attacker_name: string;

    defender_id: string;
    defender_name: string;

    winner_id: string | null;
    winner_name: string;

    damage_dealt: number;

    attacker_hp_final: number;
    defender_hp_final: number;

    rounds_count: number;
    rounds: BattleRound[];

    created_at: string;
}

export interface BattleHistory {
    id: string;

    attacker_id: string;
    attacker_name: string;

    defender_id: string;
    defender_name: string;

    winner_id: string | null;
    winner_name: string;

    damage_dealt: number;

    attacker_hp_final: number;
    defender_hp_final: number;

    rounds_count: number;

    created_at: string;
}